### DISCUSSION TITLE
使用mysql client时，使用hint不生效

### DISCUSSION THREAD

**Discussion by PanternBao** [actiontech/dble#2951] [Q&A] [answered]
使用mysql clinet 下发带有hint的sql语句到指定节点，没有生效

**Comment by PanternBao** [comment, marked as answer]
dble中 hint 使用有两种方式：

1. `/*!dble:sql=select 1 from sharding_4_t1*/ sql` 
2. `/*#dble:sql=select 1 from sharding_4_t1*/ sql` 

使用第一种方式不会遇到此种情况，而第二种方式可能会不生效。

在这里需要介绍一下mysql clinet登录时的 **_-c_** 选项：
>  -c, --comments       
>  Preserve comments. Send comments to the server. 
>  The default is --skip-comments (discard comments), enable with --comments.

当使用 -c 选项后，mysql client 会保留命令中的注释，否则会吃掉注释，因此/*# hint */ 包裹的hint不会下发到dble中，从而造成了hint失效。

### 解决方法
在mysql client 中登录时添加-c选项。

**Reply by yanhuqing666** [reply]
解释下-c的含义，另外只有一种注释会被skip
